frame/bounties/src/lib.rs

```diff
@@ -653,12 +780,596 @@ decl_module! {
+									} else {
+										// Curator has more time to give an update.
+										return Err(Error::<T>::Premature.into())
+									}
+								} else if sender == master_curator {
+									// looks like subcurator is inactive, slash the deposit.
+									slash_curator(subcurator, &mut subbounty.curator_deposit);
+									// Continue to change bounty status below...
+								} else {
+									// Else this is the subcurator or master curator,
+									// willingly giving up their role.
+									// Give back their deposit.
+									T::Currency::unreserve(&subcurator,
+										subbounty.curator_deposit,
+									);
+									// Continue to change bounty status below...
+								}
+							},
+						}
+					},
+					SubBountyStatus::PendingPayout { ref subcurator, .. } => {
+						// TODO :: Have to recheck
+						// The subbounty is in pending payout state,
+						// only Root or Master curator origin can unassign.
+						// By doing so, they are claiming the subcurator
+						// is acting maliciously, so We slash the subcurator.
+						ensure!(maybe_sender
+							.map_or(true, |sender| sender == master_curator),
+							BadOrigin
+						);
+						slash_curator(subcurator, &mut subbounty.curator_deposit);
+						// Continue to change bounty status below...
+					},
+				};
+				// Move the subbounty state to Added.
+				subbounty.status = SubBountyStatus::Added;
+				Ok(())
+			})?;
+		}
+
+		/// Award subbounty to a beneficiary.
+		///
+		/// The beneficiary will be able to claim the
+		/// funds after a delay.
+		///
+		/// The dispatch origin for this call must be
+		/// the master curator or subcurator of this subbounty.
+		///
+		/// Parent bounty must be in active state,
+		/// for this subbounty call to work.
+		///
+		/// Subbounty must be in active state, for
+		/// processing the call. and state of subbounty is
+		/// moved to PendingPayout on successful call
+		/// completion.
+		///
+		/// - `bounty_id`: ID pair Bounty ID.
+		/// - `subbounty_id`: ID pair SubBounty ID to cancel.
+		/// - `beneficiary`: Beneficiary account.
+		#[weight = 10_000]
+		fn award_subbounty(origin,
+			#[compact] bounty_id: BountyIndex,
+			#[compact] subbounty_id: BountyIndex,
+			beneficiary: <T::Lookup as StaticLookup>::Source
+		) {
+			let signer = ensure_signed(origin)?;
+			let beneficiary = T::Lookup::lookup(beneficiary)?;
+
+			// Ensure parent bounty is Active
+			let ( master_curator, _) = Self::ensure_bounty_active(bounty_id)?;
+
+			// Ensure subbounty is in expected state
+			SubBounties::<T>::try_mutate_exists(bounty_id, subbounty_id,
+				|maybe_subbounty| -> DispatchResult {
+				let mut subbounty = maybe_subbounty
+					.as_mut()
+					.ok_or(Error::<T>::InvalidIndex)?;
+
+				// Ensure Subbounty is in active state
+				match &subbounty.status {
+					SubBountyStatus::Active {
+						subcurator,
+						..
+					} => {
+						// TODO :: Have to recheck
+						// Can master curator award subbounty
```

Comment:
maybe not. only allow subcurator to award for now (again, keep thing simple), we can always change it later
also parent curator can always unasign subcurator and assign to their own and award

Reply:
Sure, Thanks. Changes are done. Only subcurator can award the subbounty.